Sources/PackageManagerDocs/Documentation.docc/Dependencies/PackageTraits.md

```diff
@@ -0,0 +1,138 @@
+
+> Note: Traits should always *enable* additional code, never "remove" or disable API when a trait is enabled.
+
+Within the package, traits express conditional compilation, and may be used to declare additional dependencies that are enabled when that trait is active.
+
+Traits are identified by their names, which are name-spaced within the package that hosts them.
+Trait names must start with a [Unicode XID start character](https://unicode.org/reports/tr31/#Figure_Code_Point_Categories_for_Identifier_Parsing) (most letters), a digit, or `_`, with subsequent characters a [Unicode XID continue character](https://unicode.org/reports/tr31/#Figure_Code_Point_Categories_for_Identifier_Parsing) or the characters `-` or `+`.
+The trait names `default` and `defaults` (regardless of any capitalization) aren't allowed to avoid confusion with the default traits that a package defines.
+
+Enabled traits are exposed as conditional blocks (for example, `#if YourTrait`) that you can use to conditionally enable imports or different compilation paths in code.
+
+## Overview
+
+### Declaring Traits
+
+Create a trait to define a discrete amount of additional functionality, and define it in the [traits](https://docs.swift.org/swiftpm/documentation/packagedescription/package/traits) property of the package manifest.
+Use [`.default(enabledTraits:)`](https://docs.swift.org/swiftpm/documentation/packagedescription/trait/default(enabledtraits:)) to provide the set of traits that the package uses as a default.
+If you don't define a default set of traits to enable, no traits are enabled by default.
+
+The following example illustrates a single trait, `FeatureA`, that is enabled by default:
+
+```swift
+// ...
+traits: [
+    .trait(name: "FeatureA"),
+    .default(enabledTraits: ["FeatureA"]),
+],
+// ...
+```
+
+Traits may also be used to represent a set of other traits, which allows you to group features together.
+The following example illustrates defining three traits, and an additional trait (`B-and-C`) that enables both traits `FeatureB` and `FeatureC`:
+
+```swift
+// ...
+traits: [
+    .trait(name: "FeatureA"),
+    .trait(name: "FeatureB"),
+    .trait(name: "FeatureC"),
+    .trait(name: "B-and-C", enabledTraits: ["FeatureB", "FeatureC"]).
+    .default(enabledTraits: ["FeatureA"]),
+],
+// ...
+```
+
+The traits enabled by default for the example above is `FeatureA`.
+
+> Note: Changing the default set of traits for your package should be considered a major semantic version change, as it can potentially remove API surface. 
+
+#### Mutually Exclusive Traits
+
+The package manifest format doesn't support declaring mutually exclusive traits.
+In the rare case that you need to offer mutually exclusive traits, protect that scenario in code:
```

Comment:
Can we add an additional line of caution here that this can result in compilation errors when users of the package enable mutually exclusive traits.